docs/deployment-file-v2.md

```diff
@@ -71,8 +71,32 @@ post: post.php
 
 When developing pre/post scripts, it helps if you are able to execute them once and see what they do. To that end you can use [remote SSH commands](/remote-ssh-execution-pro).
 
+Bear in mind the pre script won't have access to dependencies/libraries during your first deployment, since `composer install` hasn't ran at that time. Subsequent deployments will have those files however, since the `/vendor` directory is sustained - unless you reset the repositiory.
+
+Only a single pre/post script will be executed, chaining won't work. If you need to run multiple scripts, you'll need to create a wrapper script and require your other scripts within.
+
 ## Multi staging use case
 
 To allow using a single git repo with [multiple remotes](multi-staging) you can use an App name based deployment file name.
 
 Assuming you are using two Apps, `your-app-prod` and `your-app-stage`, you can setup two deployment files with named: `fortrabbit.your-app-prod.yml` and `fortrabbit.your-app-stage.yml`. This way, you can have both deployment files in a single repo but which one is to be used is determined by the App you deploy to.
+
+## Private composer use case
+
+The need for a pre script is rare, but if you [use private repositories](/private-composer-repos) over SSH (i.e. rather than `github-oauth`) and are concerned about a (highly unlikely) MITM attack, you can add a simple pre script as follows:
+
+```php
+<?php
+// Prevent MITM, by enabling StrictHostKeyChecking and adding a verified host for github.com before composer SSH installs
```

Comment:
edge case?

Reply:
100% agree! We got the [original code in a gist](https://gist.github.com/ukautz/3b798fde25e38a4d05476d0c2675a105) through a support chat so figured was something that has been asked before and may be helpful, but you guys will know better than I whether it's useful enough to include here...

This originated from https://help.fortrabbit.com/private-composer-repos recommending SSH keys for private repositories, would you suggest `github-oauth` in `composer.json` instead, which I assume would eliminate this edge case?